src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php

```diff
@@ -189,15 +189,27 @@ protected function addCookieToResponse($request, $response)
         }
 
         $response->headers->setCookie(
-            new Cookie(
-                'XSRF-TOKEN', $request->session()->token(), $this->availableAt(60 * $config['lifetime']),
-                $config['path'], $config['domain'], $config['secure'], false, false, $config['same_site'] ?? null
-            )
+            $this->newCookie($request, $config)
         );
 
         return $response;
     }
 
+    /**
+     * Create the CSRF token that will be sent with the response.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Illuminate\Config\Repository
+     * @return \Symfony\Component\HttpFoundation\Cookie
+     */
+    protected function newCookie($request, $config)
+    {
+        return new Cookie(
+            'XSRF-TOKEN', $request->session()->token(), $this->availableAt(60 * $config['lifetime']),
+            $config['path'], $config['domain'], $config['secure'], false, false, $config['same_site'] ?? null
+        );
+    }
```

Comment:
Why not just passing the request, since the only usage of the method is this one, you could just get the config in there, if missing? (Then you could remove `src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php:185`)

```suggestion
    protected function newCookie($request, $config = null)
    {
        $config = $config ?: config('session');

        return new Cookie(
            'XSRF-TOKEN', $request->session()->token(), $this->availableAt(60 * $config['lifetime']),
            $config['path'], $config['domain'], $config['secure'], false, false, $config['same_site'] ?? null
        );
    }
```